regression/failed-tests-printer.pl

```diff
@@ -6,20 +6,21 @@
 
 my $printed_this_test = 1;
 my $current_test = "";
+my $output_file = "";
 
 while (<LOG>) {
   chomp;
   if (/^Test '(.+)'/) {
     $current_test = $1;
     $printed_this_test = 0;
+  } elsif (/Output:\s+([^\s]+)/) {
+    $output_file = $1;
   } elsif (/\[FAILED\]\s*$/) {
     if(0 == $printed_this_test) {
       $printed_this_test = 1;
       print "\n\n";
       print "Failed test: $current_test\n";
-      my $outf = `sed -n '2p' $current_test/test.desc`;
-      $outf =~ s/\..*$/.out/;
-      system("cat $current_test/$outf");
+      system("cat $current_test/$output_file");
       print "\n\nFailed test.desc lines:\n";
```

Comment:
I'm not sure what you do mean by the "test descriptor". "foo.desc" will spawn "foo.out" in the same directory, "bar.desc" -> "bar.out" etc. Should I go through all repositories and rename test-name/test.desc to test-name/test-name.desc? That seems like a structural change and would require a separate ticket.